ASTInfoGenerator/DeclParser/DeclParser.py

```diff
@@ -0,0 +1,382 @@
+        if len(original_name_split) > 1 and original_name_split[1] != "":
+            original_name = original_name_split[1]
+        else:
+            original_name = original_name_split[0]
+
+        typedef_alias_name = ""
+        start_index = typedef_decl.find("referenced") + len("referenced") + 1
+        cur_char = typedef_decl[start_index]
+
+        while (cur_char != " "):
+            typedef_alias_name += cur_char
+
+            start_index += 1
+            cur_char = typedef_decl[start_index]
+
+        if original_name != typedef_alias_name:
+            return TypedefDecl(original_name, typedef_alias_name)
+
+        return None
+
+    def _parse_field_decl(self, field_decl: str, field_index: int):
+        start_index = field_decl.find("'") - 2
+        cur_char = field_decl[start_index]
+        field_name = ""
+
+        while (cur_char != " "):
+            field_name = cur_char + field_name
+
+            start_index -= 1
+            cur_char = field_decl[start_index]
+
+        return FieldDecl(field_name, field_index)
+
+    # only checks for vars whose type is struct
+    def _parse_var_decl(self, var_decl: str, specifier_manager: SpecifierManager):
+        var_decl_chunks = var_decl.split(" ")
+
+        logout(var_decl)
+        logout(var_decl_chunks)
+
+        if var_decl_chunks[5] == "used":
+            var_name = var_decl_chunks[6]
+        elif var_decl_chunks[6] == "used":
+            var_name = var_decl_chunks[7]
+        else:
+            var_name = var_decl_chunks[5]
+
+        struct_name = ""
+        start_index = var_decl.find("'") + 1
+        cur_char = var_decl[start_index]
+
+        while (cur_char != "'"):
+            struct_name += cur_char
+
+            start_index += 1
+            cur_char = var_decl[start_index]
+
+        struct_name_split = struct_name.split(" ")
+
+        # check if type is formatted like 'my_struct *' (typedef alias in use)
+        # instead of 'struct my_struct *' (no typedef alias in use)
+        type_formatted_using_typedef = len(
+            struct_name_split) > 1 and "*" in struct_name_split
+
+        if len(struct_name_split) == 2 and "*" in struct_name_split[1]:
+            struct_name = struct_name_split[0]
+        elif len(struct_name_split) > 1 and struct_name_split[1] != "" and not type_formatted_using_typedef:
+            struct_name = struct_name_split[1]
+        else:
+            struct_name = struct_name_split[0]
+
+        if not specifier_manager.struct_exists(struct_name):
+            logout(
+                f"did not find for parsed var_decl var name = {var_name} struct name = '{struct_name}'")
+            return None
+
+        logout(var_decl)
+        logout(
+            f"did parsed var_decl var name = {var_name} struct name = '{struct_name}'")
+
+        return StructVarDecl(var_name, struct_name)
+
+    def _parse_anno(self, anno_decl: str, specifier_manager: SpecifierManager, param_index: int, field_index: int):
```

Comment:
Don't we already have annotation parsing code in Python in one of your other helper files? If so, would it be possible to combine this parser and that one? If not, why not?

Reply:
(Even if only some of the code can be shared, it would be worthwhile to share it.)

Reply:
They will be combined in future where we make individual AST passes. The code I think you're referring to is `Annotations/annotation_generator.py`, and I intend on replacing it with a pass that just reads in and outputs the annotations generated by this info generator.